turbopack/crates/turbopack-core/src/resolve/mod.rs

```diff
@@ -2817,8 +2818,8 @@ async fn resolved(
     original_request: Vc<Request>,
     options_value: &ResolveOptions,
     options: Vc<ResolveOptions>,
-    query: Vc<RcStr>,
-    fragment: Vc<RcStr>,
+    query: &RcStr,
```

Comment:
nit: Isn't it better to take `RcStr`? We are always cloning it, so it may be better to take `RcStr`, if it does not make API worse

Reply:
thanks, i was actually wondering about this....

cloning is either free or an atomic increment, but it also requires a `drop` which again is either free or an atomic increment.  Now these `query` parameters are almost always `RcStr::default`, so both of these are just checking a tag bit.

but passing by reference avoids those costs... but maybe introduces other costs of indirection?  Im guessing the indirection is worse?

Is it your assumption that `&RcStr` is always a bad idea?

Reply:
My idea is that if you take `&RcStr` and always clone it, it's an unconditional call to `clone`, even when the caller has the ownership of the `RcStr`.

But this is a general Rust API guideline rather than a real problem in this case, because `RcStr` is cheap to clone, and it's mostly `RcStr::default()`.

So it's not a big deal. It's a really nitpicking.

Reply:
I appreciate the nitpicks.  Thanks!